test/helpers.ts

```diff
@@ -21,3 +29,43 @@ function fillStateWithSequence(state: ndarray.NdArray) {
         d[i] = i;
     }
 }
+
+const sqrtDoubleEpsilon = Math.pow(2, -52 / 2);
+
+export function approxEqual(x: number, y: number,
+                            tolerance = sqrtDoubleEpsilon) {
+    let xy = Math.abs(x - y);
+    const xn = Math.abs(x);
+    if (xn > tolerance) {
+        xy /= xn;
+    }
+    return xy < tolerance;
+}
+
+export function approxEqualArray(x: Float64Array, y: Float64Array,
```

Comment:
What does tolerance mean here? I would have thought it meant each number in the array had to be within the given tolerance

Reply:
The absolute or relative difference between `x` and `y` within which we'd consider them to be the same (so 1 and 1 + 1e-10 are equal, and 1e-9 and 1e-9 + 1 are also equal).

This is basically a clone of the algorithm in R's all.equal, which is not wildly different to the idea in jest's `toBeCloseTo`.

Or am I misunderstanding the question?

Reply:
Ok yeah, it was the relative difference part that I didn't expect; I always thought these type of assertions were about absolute differences